sda-download/internal/config/config.go

```diff
@@ -479,24 +482,36 @@ func constructWhitelist(obj []TrustedISS) *jwk.MapWhitelist {
 }
 
 // GetC4GHKey reads and decrypts and returns the c4gh key
-func GetC4GHKey() (*[32]byte, error) {
-	log.Info("reading crypt4gh private key")
-	keyPath := viper.GetString("c4gh.filepath")
-	passphrase := viper.GetString("c4gh.passphrase")
+func GetC4GHKey() ([32]byte, string, error) {
+	log.Info("creating temporary crypt4gh key")
+
+	public, private, err := keys.GenerateKeyPair()
 
-	// Make sure the key path and passphrase is valid
-	keyFile, err := os.Open(keyPath)
 	if err != nil {
-		return nil, err
+		log.Errorf("Error when generating keys: %v", err)
+
+		return [32]byte{}, "", err
 	}
 
-	key, err := keys.ReadPrivateKey(keyFile, []byte(passphrase))
+	pem := bytes.Buffer{}
+	err = keys.WriteCrypt4GHX25519PublicKey(&pem, public)
+
```

Comment:
```suggestion
```

Or put the entire `keys.WriteCrypt4GHX25519PublicKey(&pem, public)` in the if statement below